web/components/contract/contract-info-dialog.tsx

```diff
@@ -180,3 +203,69 @@ const getTweetText = (contract: Contract, isCreator: boolean) => {
 
   return `${tweetQuestion}\n\n${tweetDescription}\n\n${url}`
 }
+
+export function EditableResolutionTime(props: {
+  time: number
+  contract: Contract
+  isCreator: boolean
+}) {
+  const { time, contract, isCreator } = props
+
+  const [isEditing, setIsEditing] = useState(false)
+  const [timeString, setTimeString] = useState(time && formatTime(time))
+
+  const onSave = () => {
+    const newTime = dayjs(timeString).valueOf()
+    if (newTime === time) setIsEditing(false)
+    else if (contract.closeTime && newTime > (contract.closeTime ?? Date.now)) {
```

Comment:
I think you need to call `Date.now()`. Although, closeTime is always defined at the moment.